content/blog/2021/spending_policy_demo.md

```diff
@@ -0,0 +1,435 @@
+
+```bash
+bdk-cli wallet -w alice -d $ALICE_DESCRIPTOR policies
+{
+  "external": {
+    "contribution": {
+      "conditions": {
+        "0": [
+          {}
+        ],
+        "3": [
+          {
+            "csv": 2
+          }
+        ]
+      },
+      "items": [
+        0,
+        3
+      ],
+      "m": 3,
+      "n": 4,
+      "type": "PARTIAL"
+    },
+    "id": "tjdhrksc",
+    "items": [
+      {
+        "contribution": {
+          "condition": {},
+          "type": "COMPLETE"
+        },
+        "fingerprint": "5adb4683",
+        "id": "uyxvyzqt",
+        "satisfaction": {
+          "type": "NONE"
+        },
+        "type": "SIGNATURE"
+      },
+      {
+        "contribution": {
+          "type": "NONE"
+        },
+        "fingerprint": "4331a5e6",
+        "id": "w0xldflx",
+        "satisfaction": {
+          "type": "NONE"
+        },
+        "type": "SIGNATURE"
+      },
+      {
+        "contribution": {
+          "type": "NONE"
+        },
+        "fingerprint": "eebab672",
+        "id": "ghlsq5yk",
+        "satisfaction": {
+          "type": "NONE"
+        },
+        "type": "SIGNATURE"
+      },
+      {
+        "contribution": {
+          "condition": {
+            "csv": 2
+          },
+          "type": "COMPLETE"
+        },
+        "id": "8kel7sdw",
+        "satisfaction": {
+          "type": "NONE"
+        },
+        "type": "RELATIVETIMELOCK",
+        "value": 2
+      }
+    ],
+    "satisfaction": {
+      "type": "NONE"
+    },
+    "threshold": 3,
+    "type": "THRESH"
+  },
+  "internal": null
+}
+```
+
+## Step 5a: Create spending transaction
+
+The transaction can also be created by Alice, Bob, or Carol.
+
+```bash
+bdk-cli wallet -w alice -d $ALICE_DESCRIPTOR create_tx -a --to tb1qm5tfegjevj27yvvna9elym9lnzcf0zraxgl8z2:0 --external_policy "{\"tjdhrksc\": [0,1,2]}"
+{
+  "details": {
+    "fees": 169,
+    "height": null,
+    "received": 0,
+    "sent": 10000,
+    "timestamp": 1614058791,
+    "transaction": null,
+    "txid": "3b9a7ac610afc91f1d1a0dd844e609376278fe7210c69b7ef663c5a8e8308f3e"
+  },
+  "psbt": "cHNidP8BAFIBAAAAAYx7T0cL7EoUYBEU0mSL6+DS4VQafUzJgAf0Ftlbkya5AQAAAAD/////AWcmAAAAAAAAFgAU3RacollkleIxk+lz8my/mLCXiH0AAAAAAAEBKxAnAAAAAAAAIgAgCBH16JNfSPhmjJR75EdDB+gSCEF7tStNOLqw/k3BvA0BBXchA3c1Ak2kcGOzOh6eRXFKfpnpzP1lzfcXIYhxFGZG51mxrHwhA75YDXRLDLt+eX5UsE03mIGUSsQP2MrJ9lm17cGXDw2mrJN8IQIvNjaP+mwNC0DtgaB6ENB/DPPlbUDR6+NZ4Sw070jzOKyTfHZjUrJpaJNThyIGAi82No/6bA0LQO2BoHoQ0H8M8+VtQNHr41nhLDTvSPM4DO66tnIAAAAAAAAAACIGA3c1Ak2kcGOzOh6eRXFKfpnpzP1lzfcXIYhxFGZG51mxGFrbRoNUAACAAQAAgAAAAIAAAAAAAAAAACIGA75YDXRLDLt+eX5UsE03mIGUSsQP2MrJ9lm17cGXDw2mDEMxpeYAAAAAAAAAAAAA"
+}
+
+export ALICE_PSBT=$(bdk-cli wallet -w alice -d $ALICE_DESCRIPTOR create_tx -a --to tb1qm5tfegjevj27yvvna9elym9lnzcf0zraxgl8z2:0 --external_policy "{\"tjdhrksc\": [0,1,2]}" | jq -r ".psbt")
```

Comment:
For me, this variable feels like it should be called something like UNSIGNED_PSBT, because it's the pure, empty psbt. In this step Alice doesn't actually sign it at all, she simply creates it.

Reply:
I see what you mean but my focus was more around naming it for who needs to sign it. I think your point is anyone can sign initially, which is true. Going to just leave it as is for this one.

Reply:
More descriptive names do make sense.. I'm changing to:
UNSIGNED_PSBT
ALICE_SIGNED_PSBT
ALICE_BOB_SIGNED_PSBT
FINAL_PSBT